pkgs/build-support/node/build-npm-package/hooks/npm-config-hook.sh

```diff
@@ -72,6 +72,7 @@ npmConfigHook() {
     npm config set cache "$cachePath"
     npm config set offline true
     npm config set progress false
+    npm config set node_gyp "$(dirname node)/../lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js"
```

Comment:
Is this an actual npm config option? I don't see it at https://docs.npmjs.com/cli/v9/using-npm/config

Reply:
Ah I found https://github.com/nodejs/node-gyp/blob/38f01fa57d10fdb3db7697121d957bc2e0e96508/docs/Force-npm-to-use-global-node-gyp.md but it says that it only works with npm v6 and older and links to an alternative document for 7+

Reply:
Yeah actually `node-gyp-bin` does still check `npm_config_node_gyp` so I don't see any reason this wouldn't still work, actually (it's just not something npm knows or cares about)

If you have other info that may be helpful to inform what this part of the change does, that would be nice, but otherwise I think I get what this is doing now (sorry for the noise)

Reply:
You've got the idea, but I'll write up an explanation for the patch that may clear up anything else you're confused about.